## ReadMe file

Here you can find some useful tutorials on RNAseq analyses

![image](https://user-images.githubusercontent.com/17006122/210064263-6d487b75-e7ec-4913-a367-6c46c1d3a88a.png)


## Useful links

**RNA-Seq related tools**

1. [HiSat2](https://ccb.jhu.edu/software/hisat2/)
2. [Star](https://github.com/alexdobin/STAR)
3. [Tophat](http://tophat.cbcb.umd.edu/)
4. [Bioinformatics tools](https://bioinformaticshome.com/tools/msa/descriptions/Multi-LAGAN.html#gsc.tab=0)
5. [Biotools](https://services.healthtech.dtu.dk/?Seq2Logo-2.0)
6. [Format Converter](http://genome2d.molgenrug.nl/g2d_tools_conversions.html)
7. [BedTools](https://bedtools.readthedocs.io/en/latest/content/tools/getfasta.html)
8. [OnlineAnalyzor](https://cqs-vumc.shinyapps.io/rnaseqsamplesizeweb/)
9. [GEO2R TUTORIAL](https://wiki.bits.vib.be/index.php/Analyze_GEO_data_with_GEO2R)
10. [GEO2RBI](https://wiki.bits.vib.be/index.php/Analyse_GEO2R_data_with_R_and_Bioconductor)
11. [SRA](https://www.ncbi.nlm.nih.gov/sra)
12. [WebLogo](https://github.com/WebLogo/weblogo)
13. [SequenceLogoinR](https://omarwagih.github.io/ggseqlogo/)
14. [SequenceLogoinPython](https://github.com/saketkc/pyseqlogo)
15. [gglogoinR](https://github.com/heike/gglogo)

## Full RNAseq guidelines:

1. [Informatics for RNA-seq: A web resource for analysis on the cloud](https://github.com/griffithlab/rnaseq_tutorial)



## RNAseq steps
1. [Quality control](https://github.com/zagrosman/Students/blob/master/RNAseq/Qualitycontrol.md)
2. [Trimming](https://github.com/zagrosman/Students/blob/master/RNAseq/trimming.md)
3. [Mapping]( )
